pilot/pkg/config/kube/crdclient/client.go

```diff
@@ -178,8 +178,6 @@ func NewForSchemas(client kube.Client, opts Option, schemas collection.Schemas)
 			gvk.GatewayClass:      newWaiter(),
 		},
 	}
-	_ = out.crdMetadataInformer.SetTransform(kube.StripUnusedFields)
-
 	known, err := knownCRDs(client.Ext())
```

Comment:
This is the important part that is missing in the MCS code. We need to syncronuously check if the CRDs are present at startup, so we do not start Istiod with missing resources. Then we can start async (eventually consistent) adding new CRDs being added. But if the CRD is there when Istiod is started, it MUST be processed.

Howev,er i am not sure why we are doing a List() on the API. We can just watch for crdWatcher to be synced and rely on the informer List()

Reply:
>Howev,er i am not sure why we are doing a List() on the API. We can just watch for crdWatcher to be synced and rely on the informer List()

This maybe because the informer has not been started yet when calling this in the initiation

Reply:
Ah good point. Would be good to move this into crdWatcher in the future maybe, but fine as is